app/uk/gov/hmrc/api/testlogin/config/SessionTimeoutFilterWithWhitelist.scala

```diff
@@ -16,24 +16,24 @@
 
 package uk.gov.hmrc.api.testlogin.config
 
-import org.joda.time.{DateTime, DateTimeZone, Duration}
+import akka.stream.Materializer
+import javax.inject.Inject
 import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.filters.frontend.{SessionTimeoutFilter, SessionTimeoutFilterConfig}
 
-import scala.concurrent.Future
-import uk.gov.hmrc.play.frontend.filters.{ MicroserviceFilterSupport, SessionTimeoutFilter }
+import scala.concurrent.{ExecutionContext, Future}
 
 case class WhitelistedCall(uri: String, method: String)
 
-class SessionTimeoutFilterWithWhitelist(clock: () => DateTime = () => DateTime.now(DateTimeZone.UTC),
-                                        timeoutDuration: Duration,
-                                        additionalSessionKeysToKeep: Set[String] = Set.empty,
-                                        onlyWipeAuthToken: Boolean = false, whitelistedCalls: Set[WhitelistedCall])
-  extends SessionTimeoutFilter(clock, timeoutDuration, additionalSessionKeysToKeep, onlyWipeAuthToken)
-    with Filter with MicroserviceFilterSupport {
+class SessionTimeoutFilterWithWhitelist @Inject()(config: SessionTimeoutFilterConfig)(implicit ec: ExecutionContext, override val mat: Materializer)
+  extends SessionTimeoutFilter(config) {
```

Comment:
the config param and extension feels like repetition, is there a way we could use one or the other?